gym_solo/core/rewards.py

```diff
@@ -9,6 +10,14 @@
 
 
 class Reward(ABC):  
+  """A reward for a body in a pybullet simulation.
+
+  Attributes:
+    _client: The PyBullet client for the instance. Will be set via a
+      property setter.
+  """
+  _client: bullet_client.BulletClient = None
```

Comment:
Can you explain this client business a little bit. From what I understand, each instance is going to have its respective client correct?

Reply:
Yep. The issue mostly comes in rendering. You can't switch between `DIRECT` and `GUI` for the pybullet sim, so you need to restart up a new sim if you wanna say, train *then* render your results. The client just adds a layer of encapsulation so that the same script can run two pybullet instances and render, rather than us having to create a new script every time we want to render.

Reply:
I see! Smart design